src/polymer/behavior-scanner.ts

```diff
@@ -67,11 +67,34 @@ class BehaviorVisitor implements Visitor {
   propertyHandlers: PropertyHandlers|null = null;
 
   document: JavaScriptDocument;
+
   constructor(document: JavaScriptDocument) {
     this.document = document;
   }
 
   /**
+   * Fix how comments are attached to SequenceExpressions. In espree, any
+   * leading comments to the expression are parsed as leading comments to the
+   * entire sequence and not the first expression in that sequence.
+   *
+   * For our behaviors we expect our `@polymerBehavior` comments to be the
+   * leading comments of the actual first expression, so we copy them over here
+   * before so that other enter functions can properly read them.
+   */
+  enterSequenceExpression(
+      node: estree.SequenceExpression, parent: estree.Node) {
+    node.expressions.forEach((expression, i) => {
+      if (i === 0 && !expression.leadingComments) {
+        node.leadingComments = node.leadingComments || parent.leadingComments;
```

Comment:
This can be solved by a base class that keeps a stack of ancestors. Given that we know we have an open ecosystem of visitors, I'd really discourage modifying the AST in any way. If for some reason maintaining a path stack doesn't work, a WeakMap of comments would be better. One pass to build the map, another pass to use it.